shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -108,8 +108,9 @@ public FlutterRenderer(@NonNull FlutterJNI flutterJNI) {
               public void onResume(@NonNull LifecycleOwner owner) {
                 Log.v(TAG, "onResume called; notifying SurfaceProducers");
                 for (ImageReaderSurfaceProducer producer : imageReaderProducers) {
-                  if (producer.callback != null) {
-                    producer.callback.onSurfaceCreated();
+                  if (producer.callback != null && producer.notifiedDestroy) {
```

Comment:
So this check ensures that we don't fire the callback multiple times if the surface wasn't prev destroyed. Makes sense!